### ISSUE TITLE
[Request improvement] DVD Movies Compatibility

### ISSUE THREAD

**Issue by atomic83GitHub** [PCSX2/pcsx2#1981]
It would be nice to brings DVD Movies on pcsx2 , actualy , play a DVD Movie on pcsx2 cause a black screen (but no crash)

**Comment by Farmerjoe12** [comment]
Why would you watch movies through pcsx2, when you're on a computer, likely with a CD drive, and the ability to download any other software (VLC)

**Comment by lightningterror** [comment]
The feature would have been nice if pcs didn't have the ability to watch movies xD

Anyway it's not even worth putting a git issue on this.

**Comment by atomic83GitHub** [comment]
I assume its a litle useless but it should be fun to play DVD Movies like the real ps2 :)

**Comment by Blackbird88** [comment]
It's not out of the question since even Dolphin implemented unused IOS feature, but the thing is they have the manpower to do anything they want lol
If someone was determined enough it could probably be done.

**Comment by atomic83GitHub** [comment]
After some reflections , I close this request and will reopened it when the most important improvement/issues are merged.

**Comment by mirh** [comment]
PS2 could do it, so it's absolutely within scope of the project. 
Then, depending on where CSS keys were stored on original hardware I guess things might be easier or harder. 

Also depends on #432

**Comment by Phantaztique** [comment]
Would implementing this improve performance on FMV playback?

**Comment by MrCK1** [comment]
FMV playback is handled by the IPU. What performance is there to gain lol?

**Comment by willkuer** [comment]
It could be interesting for framerates and output resolutions. Also it might help regarsing general disk management (i.e. might help resolving disk read and swap issues). Also who knows. Maybe some non-bootable games use a bios routine that is typically only used by dvd's. In general improving the accuracy of the emulator helps improving the accuracy of the emulator. Emulation of the psx on the emulated ps2 might also solve some unresolved issues.

**Comment by lightningterror** [comment]
I'll close this since it's really low priority. I don't think anyone will be looking in to this for a long time.

**Comment by mirh** [comment]
It's a feature missing of a PS2?

You could close even psx and optical audio issues then with this attitude

**Comment by lightningterror** [comment]
You could improve the emulator in a lot of different ways, the problem is we lack the manpower to do so, which is why I closed it. Improving accuracy on games for both psx and ps2 takes higher priority. 

Plus the percentage of this being used by people is very low which is why it's on the lowest priority.
If someone wants they can work on it if they want to.

**Comment by turtleli** [comment]
I don't see the harm in leaving this issue open. Doing so won't force people to work on it.

Also, I don't think the Postponed label makes sense here.

**Comment by CTurt** [comment]
Just to add my own perspective in the optimistic hope that it might motivate further work on this item: there's a particularly interesting use-case for being able to emulate the DVD video playback; it would assist with reverse engineering and allow debugging of probably the most security critical part of the entire console. If you could exploit an aspect of the DVD video player, you could potentially just burn your own PS2 homebrew games and play them on an unmodified console. I would be super interested in working on that project if I had a way to debug it (pretty much impossible to test just on the hardware by burning thousands of discs and getting no real feedback on crashes).

**Comment by tadanokojin** [comment]
@kozarovv seems up your alley
`updater 'rom1:UDNL' can't load`

Got me what it means but I'm curious about it now given [this](https://github.com/CTurt/FreeDVDBoot)

**Comment by MonJamp** [comment]
According to the [blog post](https://cturt.github.io/freedvdboot.html) in that repo, the reason PCSX2 currently doesn't support playing DVDs is because the DVD player code is encrypted. So I guess it's a case of finding where the keys might be stored in the BIOS and using that to decrypt the DVD player.

**Comment by MobCat** [comment]
2~3 years ago: No, lol what, why. Your on a computer why would you want a thing like this just use vlc.
2020: Umm yeah we need to debug a DVD player exploit and having an emulated version of that would be nice to save on burning disks and needing to track down all the real hardware and test it.

**Comment by bositman** [comment]
Or maybe we have some new devs who would like to get some experience in PCSX2 code and something relatively not complicated like this could help. If you don't have something constructive to comment, maybe you should refrain from doing so...

**Comment by Florin9doi** [comment]
I tested #4274, but it doesn't resolve this ticket

**Comment by IkuSenpai** [comment]
Games like .Hack has bonus CDs that you need to watch to get bonus content in the game. Can't even find keywords via cheat engine myself nor has anyone I asked been able to. So fixing the DVD compatibility might be a better solution. 

I use the dev/experimental builds and have tested #5236 and still not functional, ticket is still not resolved.

**Comment by Florin9doi** [comment]
Mostly resolved by #4274 + #6719.

Known issues:
- GsCrt modes 0x82 and 0x83 are incorrectly implemented and the image is cropped:
![unknown](https://user-images.githubusercontent.com/2995486/181909779-fd3b7fab-4cd2-437d-93c6-52fa5cbab1f7.png)
- Some BIOS versions released in 2002 are trapped in an infinite loop. These '02 ROM1 files works properly when paired with ROM0 files from '01 or '03.
![1](https://user-images.githubusercontent.com/2995486/181913678-042a154a-3962-4eb6-8e31-bb7256b08c3f.png)

**Comment by Florin9doi** [comment]
The cropping is observed because the [screen geometry](https://github.com/PCSX2/pcsx2/blob/9a8a3601a7458a2382c97aac7e882468aa1564b0/pcsx2/GS/GSState.cpp#L460) is determined mostly by a single parameter : [the color subcarrier](https://github.com/PCSX2/pcsx2/blob/9a8a3601a7458a2382c97aac7e882468aa1564b0/pcsx2/GS/GSState.cpp#L344).

**Comment by Florin9doi** [comment]
The screen geometry is resolved by #7602 + Show Overscan option:
![Untitled](https://user-images.githubusercontent.com/2995486/208775903-a0faddae-b660-44be-894a-e1f9151118e8.png)

**Comment by Florin9doi** [comment]
Two more issues related to DVD Player update discs:
- DVD Player Version 2.01J (PBPX-95206) installation gets stuck and doesn't finish
- The emulator crashes if any Japanese DVD player update is installed on Bios v01.00j. This can be avoided by switching to EE interpreter or by deleting `mcX:/BIEXEC-SYSTEM/osdsys.elf` file.

**Comment by Florin9doi** [comment]
#7602 was merged, but the geometry fix<sup>https://github.com/PCSX2/pcsx2/issues/1981#issuecomment-1360384968</sup> was lost

<details><summary>(image)</summary><p>

![image](https://user-images.githubusercontent.com/2995486/218283197-585a863e-06e5-4b94-b26c-42844c5e815e.png)
</p></details>

**Comment by john123903** [comment]
Without it, then It gonna say DVD player not Setup. So Can we add it? (idk if you guys want to work on it.)

**Comment by john123903** [comment]
> ![Booting PS2 BIOS _20221018190406](https://user-images.githubusercontent.com/116039192/196510179-c3b04b21-ca79-429b-8e5b-ee832d1c069d.png) @Florin9doi I think I have it working with SCPH-90001 but the image is shifted to the right.

HOW DO YOU DO THAT!? It did not work on mine.

**Comment by dobkun** [comment]
and this is why refractions should finally implement MagicGate(but only the support without any documentation):
UTILITY DISCS!

**Comment by stenzek** [comment]
No. Please do not spam issues, or you will be blocked. Thanks.

**Comment by christopher3242** [comment]
Okay this good .
